packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts

```diff
@@ -0,0 +1,116 @@
+import { CAL_URL } from "@calcom/lib/constants";
+import { prisma } from "@calcom/prisma";
+import type { Webhook } from "@calcom/prisma/client";
+import { MembershipRole } from "@calcom/prisma/enums";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+import { TRPCError } from "@trpc/server";
+
+type GetByViewerOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+};
+
+type WebhookGroup = {
+  teamId?: number | null;
+  profile: {
+    slug: string | null;
+    name: string | null;
+    image?: string;
+  };
+  metadata?: {
+    readOnly: boolean;
+  };
+  webhooks: Webhook[];
+};
+
+export type WebhooksByViewer = {
+  webhookGroups: WebhookGroup[];
+  profiles: {
+    readOnly?: boolean | undefined;
+    slug: string | null;
+    name: string | null;
+    image?: string | undefined;
+    teamId: number | null | undefined;
+  }[];
+};
+
+export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
+  const user = await prisma.user.findUnique({
+    where: {
+      id: ctx.user.id,
+    },
+    select: {
+      username: true,
+      avatar: true,
+      name: true,
+      webhooks: true,
+      teams: {
+        where: {
+          accepted: true,
+        },
+        select: {
+          role: true,
+          team: {
+            select: {
+              id: true,
+              name: true,
+              slug: true,
+              members: {
+                select: {
+                  userId: true,
+                },
+              },
+              webhooks: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  if (!user) {
+    throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
+  }
+
+  const userWebhooks = user.webhooks;
+  let webhookGroups: WebhookGroup[] = [];
+
+  webhookGroups.push({
+    teamId: null,
+    profile: {
+      slug: user.username,
+      name: user.name,
+      image: user.avatar || undefined,
+    },
+    webhooks: userWebhooks,
+    metadata: {
+      readOnly: false,
+    },
+  });
+
+  const teamWebhookGroups: WebhookGroup[] = user.teams.map((membership) => ({
+    teamId: membership.team.id,
+    profile: {
+      name: membership.team.name,
+      slug: "team/" + membership.team.slug,
+      image: `${CAL_URL}/team/${membership.team.slug}/avatar.png`,
+    },
+    metadata: {
+      readOnly: membership.role === MembershipRole.MEMBER,
```

Comment:
```suggestion
      readOnly: membership.role !== MembershipRole.ADMIN && membership.role !== MembershipRole.OWNER ,
```

It makes more sense and gives better security IMHO as let's say if a new  role with lower access level then ADMIN, OWNER is added, it would automatically have readOnly false which we would not want.